lib/recipe_box/cuisine.ex

```diff
@@ -0,0 +1,46 @@
+defmodule RecipeBox.Cuisine do
+  @moduledoc """
+  This is the Cuisine context
+  """
+
+  use Ecto.Schema
+  import Ecto.Changeset
+  
+  alias RecipeBox.Cuisine
+
+  @doc "RecipeBox.Repo is the application's Ecto repository"
+  alias RecipeBox.Repo
+  
+  @doc """
+  This specifies the database table and corresponding Elixir struct at the same time.
+  Generate a migration file by running mix ecto.gen.migration create_cuisines.
+  """
+  schema "cuisines" do
+    field :name, :string
+
+    timestamps()
+  end
+  
+  @doc """
+  Changesets allow Ecto to manage record changes.
+  The cast function allows only the specified fields into the DB and nothing else.
+  """
+  def changeset(cuisine, attrs) do
+    cuisine
+    |> cast(attrs, [:name])
+    |> validate_required([:name])
+    |> downcase_value
+    |> unique_constraint(:name)
+  end
+
+  def downcase_value(changeset) do
```

Comment:
Small naming quibble: at some point, the `cuisines` table will probably have some columns on it besides `name`. I would either make this function `downcase_name` OR have it take in the field to change, like `downcase_value(changeset, field)`.

Reply:
That's a really good point! I'll edit that